readme.md

```diff
@@ -1,12 +1,105 @@
-# MongoDB Node.js Callback Legacy Package
+# MongoDB Node.js Driver with Optional Callback Support Legacy Package
 
-# Work In Progress - Not intended for use at this time
+**Attention :memo:**
 
-Here is the legacy package for callback support in the MongoDB Driver. Similar to how Node.js has begun to ship promise alternatives to our well known stdlib APIs:
+This is a wrapper of the `mongodb` driver, if you are starting a new project you likely just want to use the driver directly:
 
-```js
-const fs = require('fs/promises')
-await fs.readFile('...')
+- [Driver Source](https://github.com/mongodb/node-mongodb-native/)
+- [Driver NPM Package](https://www.npmjs.com/package/mongodb)
+
+## Purpose
+
+This package is intended to assist in migrating to promise based APIs.
+We have wrapped every driver method to continue offering the optional callback support some projects may be relying on to incrementally migrate to promises.
+Any new APIs added to the driver will not offer optional callback support.
+If callback usage is needed for any new APIs, we would suggest using `.then`/`.catch` or node's [callbackify](https://nodejs.org/dist/latest-v16.x/docs/api/util.html#utilcallbackifyoriginal) to get that desired API.
+
+The main driver package `mongodb` will be dropping optional callback support in the next major version (v5) in favor of a pure `async`/`await` syntax.
+
```

Comment:
Can we add a description or a title for this code block?  It's not clear what it is indicating without reading the code (and from line 17, it seems like it should be demonstrating the pure async-await syntax for the driver).

Reply:
done